.travis.yml

```diff
@@ -3,42 +3,98 @@ language: go
+      #go: 1.8.x
+#env:
+  ## Flip bit to bypass tests - see dep#501 for more information
+  #- DEPTESTBYPASS501=1
+#install:
+  #- echo "This is an override of the default install deps step in travis."
+#before_script:
+  ## OSX as of El Capitan sets an exit trap that interacts poorly with our
+  ## set -e below. So, unset the trap.
+  ## Related: https://superuser.com/questions/1044130/why-am-i-having-how-can-i-fix-this-error-shell-session-update-command-not-f
+  #- if [[ "$(go env GOHOSTOS)" == "darwin" ]]; then trap EXIT; fi
+  #- if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  #- if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install bzr; fi
+  #- PKGS=$(go list ./... | grep -v /vendor/ | grep -v _testdata/ )
+  #- go get -v honnef.co/go/tools/cmd/{gosimple,staticcheck}
+  #- npm install -g codeclimate-test-reporter
+#script:
+  #- go build -v ./cmd/dep
+  #- go vet $PKGS
+  #- staticcheck $PKGS
+  ##- ./hack/validate-gofmt.bash
+  #- ./hack/validate-vendor.bash
+  #- gosimple $PKGS
+  ##- go test -race $PKGS
+  #- go build ./hack/licenseok
+  #- set -e; for pkg in $PKGS; do go test -race -coverprofile=profile.out -covermode=atomic $pkg; if [[ -f profile.out ]]; then cat profile.out >> coverage.txt; rm profile.out; fi; done
+  #- find . -path ./vendor -prune -o -type f -name "*.go" -printf '%P\n' | xargs ./licenseok
+  #- ./hack/validate-vendor.bash
+#after_success:
+  #- codeclimate-test-reporter < coverage.txt
+
+jobs:
   include:
-    - os: linux
-      go: 1.7.x
-    - os: linux
+    - stage: precheck
+      install:
+        - go get -u honnef.co/go/tools/cmd/{gosimple,staticcheck}
+      os: linux
+      go: 1.8.x
+      script:
+        - go build -v ./cmd/dep
+        - PKGS=$(go list ./... | grep -v /vendor/)
+        - go vet $PKGS
+        - staticcheck $PKGS
+        - gosimple $PKGS
+        - ./hack/validate-vendor.bash
+        - go build ./hack/licenseok
+        - find . -path ./vendor -prune -o -type f -name "*.go" -printf '%P\n' | xargs ./licenseok
+    - stage: test
+      os: osx
       go: 1.8.x
-    - os: linux
+      env:
+        - DEPTESTBYPASS501=1
+      install:
+        # OSX as of El Capitan sets an exit trap that interacts poorly with our
+        # set -e below. So, unset the trap.
+        # Related: https://superuser.com/questions/1044130/why-am-i-having-how-can-i-fix-this-error-shell-session-update-command-not-f
+        - trap EXIT
+        - brew install bzr
+      script: go test -race $(go list ./... | grep -v vendor)
+    - stage: test
+      install: skip
+      env:
+        - DEPTESTBYPASS501=1
+      os: linux
       go: tip
-    - os: osx
+      script: go test -race $(go list ./... | grep -v vendor)
+    - stage: test
+      install: skip
+      os: linux
+      go: 1.7.x
+      env:
+        - DEPTESTBYPASS501=1
+      script: go test -race $(go list ./... | grep -v vendor)
+    - stage: test
+      cache: yarn
+      install: yarn global add codeclimate-test-reporter
```

Comment:
yeah it was bugging out on this - https://docs.travis-ci.com/user/caching/#yarn-cache made me think i could do without but nope